### PR TITLE
[new release] ocaml-lsp-server, lsp and jsonrpc (1.3.0)

### PR DESCRIPTION
**Pull request by rgrinberg**
LSP Server for OCaml

- Project page: <a href="https://github.com/ocaml/ocaml-lsp">https://github.com/ocaml/ocaml-lsp</a>

## Features

- Code action to insert inferred module interface (ocaml/ocaml-lsp#308)

- Filter keywords by context (ocaml/ocaml-lsp#307)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
There is an issue from OCaml 4.06 to 4.10:
```
#=== ERROR while compiling ocaml-lsp-server.1.3.0 =============================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/ocaml/ocaml-lsp/releases/download/1.3.0/jsonrpc-1.3.0.tbz)
# path                 ~/.opam/4.06/.opam-switch/build/ocaml-lsp-server.1.3.0
# command              ~/.opam/4.06/bin/dune build -j 47 ocaml-lsp-server.install --release
# exit-code            1
# env-file             ~/.opam/log/ocaml-lsp-server-1-bd5e5c.env
# output-file          ~/.opam/log/ocaml-lsp-server-1-bd5e5c.out
### output ###
# File "_build/.dune/default/ocaml-lsp-server/vendor/merlin/src/ocaml/utils/dune", line 2, characters 13-25:
# 2 | (copy_files# 406/*.ml{,i})
#                  ^^^^^^^^^^^^
# Error: Cannot find directory:
# ocaml-lsp-server/vendor/merlin/src/ocaml/utils/406
```

**Comment by kit-ty-kate** [comment]
New error on the same set of compilers:
```
#=== ERROR while compiling ocaml-lsp-server.1.3.0 =============================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/ocaml/ocaml-lsp/releases/download/1.3.0/jsonrpc-1.3.0.tbz)
# path                 ~/.opam/4.06/.opam-switch/build/ocaml-lsp-server.1.3.0
# command              ~/.opam/4.06/bin/dune build -j 47 ocaml-lsp-server.install --release
# exit-code            1
# env-file             ~/.opam/log/ocaml-lsp-server-1-53ea5a.env
# output-file          ~/.opam/log/ocaml-lsp-server-1-53ea5a.out
### output ###
#     ocamlopt ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native/mreader_lexer.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.06/bin/ocamlopt.opt -open Merlin_utils -w -a -g -I ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/byte -I ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native -I ocaml-lsp-server/vendor/merlin/src/config/.config.objs/byte -I ocaml-lsp-server/vendor/merlin/src/config/.config.objs/native -I ocaml-lsp-server/vendor/merlin/src/dot-merlin/dot-protocol/.merlin_dot_protocol.objs/byte -I ocaml-lsp-server/vendor/merlin/src/dot-merlin/dot-protocol/.merlin_dot_protocol.objs/native -I ocaml-lsp-server/vendor/merlin/src/extend/.merlin_extend.objs/byte -I ocaml-lsp-server/vendor/merlin/src/extend/.merlin_extend.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/merlin_specific/.merlin_specific.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/merlin_specific/.merlin_specific.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/parsing/.parsing.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/parsing/.parsing.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/preprocess/.preprocess.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/preprocess/.preprocess.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/typing/.typing.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/typing/.typing.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/utils/.utils.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/utils/.utils.objs/native -I ocaml-lsp-server/vendor/merlin/src/utils/.merlin_utils.objs/byte -I ocaml-lsp-server/vendor/merlin/src/utils/.merlin_utils.objs/native -intf-suffix .ml -no-alias-deps -o ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native/mreader_lexer.cmx -c -impl ocaml-lsp-server/vendor/merlin/src/kernel/mreader_lexer.ml)
# File "ocaml-lsp-server/vendor/merlin/src/kernel/mreader_lexer.ml", line 93, characters 28-52:
# Error: Unbound value Lexer_raw.ocaml_keywords
```

**Comment by rgrinberg** [comment]
My attempt to upgrade merlin was a little premature. Let me try again

**Comment by mseri** [comment]
Unfortunately it is still failing with the missing folders:
```
#=== ERROR while compiling ocaml-lsp-server.1.3.0 =============================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/ocaml/ocaml-lsp/releases/download/1.3.0/jsonrpc-1.3.0.tbz)
# path                 ~/.opam/4.06/.opam-switch/build/ocaml-lsp-server.1.3.0
# command              ~/.opam/4.06/bin/dune build -j 47 ocaml-lsp-server.install --release
# exit-code            1
# env-file             ~/.opam/log/ocaml-lsp-server-1-53ea5a.env
# output-file          ~/.opam/log/ocaml-lsp-server-1-53ea5a.out
### output ###
# File "_build/.dune/default/ocaml-lsp-server/vendor/merlin/src/ocaml/utils/dune", line 2, characters 13-25:
# 2 | (copy_files# 406/*.ml{,i})
#                  ^^^^^^^^^^^^
# Error: Cannot find directory:
# ocaml-lsp-server/vendor/merlin/src/ocaml/utils/406
```
(similarly for 4.07 -- 4.10)

**Comment by mseri** [comment]
Looks good now, thanks!
